src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -1974,6 +1993,14 @@ void Unit::AttackerStateUpdate(Unit* victim, WeaponAttackType attType, bool extr
         CalculateMeleeDamage(victim, 0, &damageInfo, attType);
         // Send log damage message to client
         DealDamageMods(victim, damageInfo.damage, &damageInfo.absorb);
+
+        // Sparring Checks
+        if (Creature* me = ToCreature())
+            if (Creature* target = victim->ToCreature())
+                if (me->CanSparWith(target))
+                    if (target->GetHealthPct() <= me->GetSparringHealthLimitPctFor(target))
+                        damageInfo.HitInfo |= HITINFO_FAKE_DAMAGE;
```

Comment:
Instead of modifying Unit::AttackerStateUpdate use Unit::FakeAttackerStateUpdate which does just this (sparring vs other creature should be checked elsewhere, for example at AI level)

Reply:
No, it doesn't. The fake attacker state sends a hacky packet. According to sniffs the damage etc. does not get modified, only this hitinfo flag gets added. The client will know then to not use the damage. The rest must be handled by the core as I did.

Reply:
Then feel free to add damage information to fake_damage but iirc I saw sniffs with hitinfo fake and 0 damage

Reply:
At that point, would there be sufficient difference between `AttackerStateUpdate` and `FakeAttackerStateUpdate` to support keeping them separate?

I am very much not fond of the code duplication involved here.

Reply:
I will review this, when I am back home. in two weeks. I will probably add more fields to damageinfo so we dont have to call the checks multiple times. This might make the old fakedamage features obsolete so I can drop them afterwards